rest/api.go

```diff
@@ -297,7 +301,13 @@ func (h *handler) handleProfiling() error {
 
 	if isCPUProfile {
 		base.Infof(base.KeyAll, "Starting CPU profile to %s ...", base.UD(params.File))
-		err = pprof.StartCPUProfile(f)
+		if err = pprof.StartCPUProfile(f); err != nil {
+			if fileError := os.Remove(params.File); fileError != nil {
+				base.Infof(base.KeyAll, "Error removing file: %s", base.UD(params.File))
+			}
+			return err
+		}
+		h.server.SetCpuPprofFile(nil)
```

Comment:
Shouldn't this parameter be `f`?

Reply:
You are right. Change is made in latest revision.